lib/puppet/provider/iis_application_pool/webadministration.rb

```diff
@@ -24,6 +24,7 @@ def create
     result   = self.class.run(inst_cmd)
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:exitcode] == 0
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:errormessage].nil?
+    @resource.provider = self.class.instances.find{|prov| prov.name == @resource[:name]}
```

Comment:
`puppet resource iis_application_pool <name>` when 4 pools exist takes 11 seconds
`puppet resource iis_application_pool <name>` when 150ish pools exist takes 12 seconds
`puppet resource iis_application_pool <name>` when 300ish pools exist takes 13 seconds
`puppet resource iis_application_pool <name> ensure=present` when 4 pools exist takes 15 seconds
`puppet resource iis_application_pool <name> ensure=present` when 150ish pools exist takes 17 seconds
`puppet resource iis_application_pool <name> ensure=present` when 300ish pools exist takes 20 seconds

So it does take a few seconds longer to create as more pools exist, but only 5 seconds longer at 300 pools than 3 pools.